docs/reference/quickstart/full-text-tutorial.asciidoc

```diff
@@ -0,0 +1,435 @@
+
+A <<query-dsl-bool-query,`bool`>> query combines multiple queries using boolean logic.
+
+[source,console]
+----
+GET /cooking_blog/_search
+{
+  "query": {
+    "bool": {
+      "must": [ <1>
+        { "match": { "description": "recipe" } }
+      ],
+      "filter": [ <2>
+        { "term": { "category.keyword": "Breakfast" } },
+        { "range": { "rating": { "gte": 4.5 } } }
+      ]
+    }
+  }
+}
+----
+// TEST[continued]
+<1> The `must` clause requires the `description` field to match "recipe". This means that only documents containing the word "recipe" in the `description` field will be considered.
+<2> The `filter` clauses ensure only documents in the "Breakfast" category with a rating of 4.5 or higher are returned, without affecting the relevance score.
+
+[discrete]
+[[full-text-tutorial-fuzzy-query]]
+=== Handle typos and variations in search terms
+
+A <<query-dsl-fuzzy-query,`fuzzy` query>> finds terms that are similar to the search term, accounting for typos or slight variations.
+Fuzzy queries have no analysis phase.
+
+[source,console]
+----
+GET /cooking_blog/_search
+{
+  "query": {
+    "fuzzy": {
+      "description": {
+        "value": "pancaeks", <1>
+        "fuzziness": "AUTO" <2>
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+<1> The term to search for, allowing for variations.
+<2> Automatically determines the appropriate level of fuzziness.
+
+[discrete]
+[[full-text-tutorial-prefix-query]]
+=== Search by prefixes
+
+A <<query-dsl-prefix-query,`prefix`>> query enables you to quickly narrow down results to documents that begin with a specified prefix.
+Note that prefix queries don't perform text analysis on search terms!
+
+[source,console]
+----
+GET /cooking_blog/_search
+{
+  "query": {
+    "prefix": { 
+      "title": {
+        "value": "Spicy",
+        "case_insensitive": true <1>
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+<1> Because the `title` field is analyzed at index time, it's been processed by the `lowercase` filter. The prefix query isn't analyzed and would return zero hits if you passed "Spicy" here, without setting `case_insensitive` to `true`.
+
+[discrete]
+[[full-text-tutorial-query-string]]
+=== Perform advanced searches with query strings
+
+A <<query-dsl-query-string-query,`query_string`>> query supports complex queries with a compact syntax.
+Use `query_string` for advanced, multi-field searches with complex syntax.
+
+[source,console]
+----
+GET /cooking_blog/_search
+{
+  "query": {
+    "query_string": {
+      "fields": ["title", "description", "tags"], <1>
+      "query": "(vegetarian OR vegan) AND (curry OR *fry) AND rating:>=4.5" <2>
+    }
+  }
+}
+----
+// TEST[continued]
+<1> Searches across multiple fields simultaneously
+<2> Combines boolean logic, wildcards, and `range` queries:
+   - `(vegetarian OR vegan)`: Matches either dietary preference
+   - `(curry OR *fry)`: Uses wildcard for various cooking methods
+   - `rating:>=4.5`: Filters for highly-rated recipes
```

Comment:
There's a lot in this example, we should expand the description

Reply:
might be one for more advanced follow-up